kernel/read.c

```diff
@@ -223,6 +294,9 @@ NTSTATUS DeviceRead(IN PDEVICE_EXTENSION extension,
 	
     if (bytes_read==0) 
 	{
+		// As it is now, the issue is that we do not know whether a real error happened or 'only' a VSM/Hyper-v induced read error.
+		// The read handler function returns either the number of bytes or 0 (but no status).
+		// We could avoid that by giving a ULONG * bytes_read to the read handler function and have a NTSTATUS returned instead. 

```

Comment:
maybe we want to change the reader to return an actual status? although realistically the only thing it can return is io error anyway.

Reply:
That is what I wrote in line 297. I was reluctant to directly change the function prototypes just like that without at least asking. That seemed like hybris.